src/main/java/tutoraid/logic/commands/EditStudentCommand.java

```diff
@@ -0,0 +1,208 @@
+package tutoraid.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static tutoraid.logic.parser.CliSyntax.PREFIX_PARENT_NAME;
+import static tutoraid.logic.parser.CliSyntax.PREFIX_PARENT_PHONE;
+import static tutoraid.logic.parser.CliSyntax.PREFIX_STUDENT_NAME;
+import static tutoraid.logic.parser.CliSyntax.PREFIX_STUDENT_PHONE;
+import static tutoraid.model.Model.PREDICATE_SHOW_ALL_STUDENTS;
+
+import java.util.List;
+import java.util.Optional;
+
+import tutoraid.commons.core.Messages;
+import tutoraid.commons.core.index.Index;
+import tutoraid.commons.util.CollectionUtil;
+import tutoraid.logic.commands.exceptions.CommandException;
+import tutoraid.model.Model;
+import tutoraid.model.student.ParentName;
+import tutoraid.model.student.PaymentStatus;
+import tutoraid.model.student.Phone;
+import tutoraid.model.student.Progress;
+import tutoraid.model.student.Student;
+import tutoraid.model.student.StudentName;
+
+/**
+ * Edits the details of an existing person in the address book.
```

Comment:
Maybe "person" can be changed to "student" here in this header comment to better fit TutorAid?
This also applies to the other JavaDoc comments in this file